src/fastga_he/powertrain_builder/unit_tests/test_power_train_file.py

```diff
@@ -218,6 +219,98 @@ def test_power_train_initialization_time():
     print(f"Get connection time: {octo_cache['get_connection_time']} sec")
 
 
+def test_load_cache():
+    sample_power_train_file_path = pth.join(pth.dirname(__file__), "data", YML_FILE)
+    power_train_configurator = FASTGAHEPowerTrainConfigurator(
+        power_train_file_path=sample_power_train_file_path
+    )
+    power_train_configurator._get_components()
+
+    load_cache = power_train_configurator._cache[sample_power_train_file_path]["_serializer"]
```

Comment:
```suggestion
    loaded_cache = power_train_configurator._cache[sample_power_train_file_path]["_serializer"]
```

Reply:
And If I may, maybe check at the beginning that the cache is empty or at least doesn't contain anything with that key :)